api/src/test/java/org/pmiops/workbench/cohortbuilder/querybuilder/util/CriteriaLookupUtilTest.java

```diff
@@ -0,0 +1,327 @@
+@Transactional
+public class CriteriaLookupUtilTest {
+
+  @Autowired private CBCriteriaDao cbCriteriaDao;
+
+  @Autowired private JdbcTemplate jdbcTemplate;
+
+  private CriteriaLookupUtil lookupUtil;
+
+  @Before
+  public void setUp() throws Exception {
+    lookupUtil = new CriteriaLookupUtil(cbCriteriaDao);
+  }
+
+  private void saveCriteriaWithPath(String path, CBCriteria criteria) {
+    cbCriteriaDao.save(criteria);
+    String pathEnd = String.valueOf(criteria.getId());
+    criteria.path(path.isEmpty() ? pathEnd : path + "." + pathEnd);
+    cbCriteriaDao.save(criteria);
+  }
+
+  @Test
+  public void buildCriteriaLookupMapDrugCriteria() throws Exception {
+    CBCriteria drugNode1 =
+        new CBCriteria()
+            .parentId(99999)
+            .domainId(DomainType.DRUG.toString())
+            .type(CriteriaType.ATC.toString())
+            .conceptId("21600002")
+            .group(true)
+            .selectable(true);
+    saveCriteriaWithPath("0", drugNode1);
+    CBCriteria drugNode2 =
+        new CBCriteria()
+            .parentId(drugNode1.getId())
+            .domainId(DomainType.DRUG.toString())
+            .type(CriteriaType.RXNORM.toString())
+            .conceptId("19069022")
+            .group(false)
+            .selectable(true);
+    saveCriteriaWithPath(drugNode1.getPath(), drugNode2);
+    CBCriteria drugNode3 =
+        new CBCriteria()
+            .parentId(drugNode1.getId())
+            .domainId(DomainType.DRUG.toString())
+            .type(CriteriaType.RXNORM.toString())
+            .conceptId("1036094")
+            .group(false)
+            .selectable(true);
+    saveCriteriaWithPath(drugNode1.getPath(), drugNode3);
+
+    // Use jdbcTemplate to create/insert data into the ancestor table
+    // The codebase currently doesn't have a need to implement a DAO for this table
+    jdbcTemplate.execute(
+        "create table cb_criteria_ancestor(ancestor_id integer, descendant_id integer)");
+    jdbcTemplate.execute(
+        "insert into cb_criteria_ancestor(ancestor_id, descendant_id) values (19069022, 19069022)");
+    jdbcTemplate.execute(
+        "insert into cb_criteria_ancestor(ancestor_id, descendant_id) values (1036094, 1036094)");
+
+    List<Long> childConceptIds = Arrays.asList(19069022L, 1036094L);
+    SearchParameter searchParameter =
+        new SearchParameter()
+            .domain(DomainType.DRUG.toString())
+            .type(CriteriaType.ATC.toString())
+            .group(true)
+            .ancestorData(true)
+            .conceptId(21600002L);
+    SearchRequest searchRequest =
+        new SearchRequest()
+            .addIncludesItem(
+                new SearchGroup()
+                    .addItemsItem(new SearchGroupItem().addSearchParametersItem(searchParameter)));
+    assertEquals(
+        ImmutableMap.of(searchParameter, new HashSet<>(childConceptIds)),
+        lookupUtil.buildCriteriaLookupMap(searchRequest));
+    cbCriteriaDao.delete(drugNode1.getId());
```

Comment:
DirtiesContext? https://github.com/all-of-us/workbench/blob/dffa11812709f42266c6dfab4c106b43bc5838c6/api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java#L97

Reply:
Added @DirtiesContext and removed the manual deletes.